### ISSUE TITLE
Don't cancel download if select not to overwrite file.

### ISSUE THREAD

**Issue by adskankster** [The-Compiler/qutebrowser#2103]
I've just tried to download a pdf ticket fromt a site that only allows single downloads of its pdf tickets (yes, I know). I didn't realise that its filenames were always the same (ticket.pdf) and tried to save it into the same directory that a previous (unused) download was in. Qutebrowser asked me if I wanted to overwrite, and I said "no", thinking that I would get the opportunity to edit the path. Instead the download just disappeared (I've tried looking under /tmp but it's not there). 

Would it be possible, when saying no to the question, to do something else with the file instead, or have (via a setting?) qutebrowser to append a suffix to prevent overwriting?

Thanks.

**Comment by The-Compiler** [comment]
Ouch! Indeed, re-offering the filename prompt would probably be the better thing to do there...

**Comment by adskankster** [comment]
Out of curiosity, where is the code for this? I had a look at the source, but couldn't find it.
Not sure I could do much, if I did see it, but...

**Comment by The-Compiler** [comment]
On the `webengine-downloads-3` branch (which moved some stuff around, and will be merge d into `master` tomorrow or so) the respective code is [here in `downloads.py`](https://github.com/The-Compiler/qutebrowser/blob/de1e3a7a54b1841faf0296599c28fbcfa62e666b/qutebrowser/browser/downloads.py#L537-L542). `_ask_confirm_question` is then implemented separately for [qtwebengine](https://github.com/The-Compiler/qutebrowser/blob/baeb8653c83d73ec7a237019e86b78c803a4a464/qutebrowser/browser/webengine/webenginedownloads.py#L107-L122) and [qtnetwork](https://github.com/The-Compiler/qutebrowser/blob/de1e3a7a54b1841faf0296599c28fbcfa62e666b/qutebrowser/browser/qtnetworkdownloads.py#L181-L189)

**Comment by adskankster** [comment]
Many thanks!

**Comment by adskankster** [comment]
Many thanks. 

I've had a quick look at qtnetwork as I don't use qtwebengine yet, and though it would take some time to work out what the no_action should be set to, doesn't the "remove_data=False" parameter mean that it shouldn't have been deleted from the /tmp directory? 

I hope to have a longer look tomorrow.

**Comment by The-Compiler** [comment]
Yeah - it's a bit tricky because the whole API is asynchronous (for QtNetwork) - I'd need to have a deeper look too to see how to implement this.

As for `remove_data=false`: No data was written to disk yet at the point the question is asked, so there's nothing to remove, but there also isn't anything there :wink:

**Comment by adskankster** [comment]
I've had a bit more of a look now and that all makes sense. I think it's a bit beyond my current experience of python and the code to do correctly. I'll work around it for now. 

Thank you for looking at it. :)

**Comment by The-Compiler** [comment]
Like pointed out in #2103, it should probably also be possible to configure that a suffix should be added to the filename.

**Comment by The-Compiler** [comment]
The same should probably apply for the "create new directory" question, see #6187
